04-onion-routing.md

```diff
@@ -601,6 +606,20 @@ If the `cltv-expiry` is too low, the final node MUST fail the HTLC:
 
 1. type: 17 (`final_expiry_too_soon`)
 
+If the `outgoing_cltv_value` does not match the `ctlv-expiry` of the
+HTLC at the final hop:
+
+1. type: 18 (`final_incorrect_cltv_expiry`)
+2. data:
+   * [4:cltv-expiry]
+
+If the `amt_to_forward` does not match the `incoming_htlc_amt` of
```

Comment:
Hm, this is introducing special handling for the last hop when it comes to the `amt_to_forward`, since this is the amount the node is allowed to keep, maybe we can flip this with the `fee`? That way validation is always identical:

- keep `fee`, i.e., full amount on last hop
- forward `amt_to_forward`, i.e., 0 on the last hop